CMakeLists.txt

```diff
@@ -3055,6 +3057,29 @@ if(SDL_CCACHE)
   endif()
 endif()
 
+if (SDL_CLANG_TIDY)
+  cmake_minimum_required(VERSION 3.6)
+  find_program(CLANG_TIDY_BINARY clang-tidy)
+
+  if(CLANG_TIDY_BINARY)
+    set(HAVE_CLANG_TIDY ON)
+    get_3rd_party_source(3RD_PARTY_SOURCES)
+    get_clang_tidy_ignored_files(CLANG_TIDY_IGNORED_FILES "${3RD_PARTY_SOURCES}")
```

Comment:
I would combine these functions into one:
```cmake
function(get_clang_tidy_ignored_files OUTVAR)
    set(sources
        "src1.c"
        ...
        "srcN.c"
    )
    # calculate json
    set(${OUTVAR} ${res} PARENT_SCOPE)
endfunction()
```
and only do the following here:
```cmake
get_clang_tidy_ignored_files(CLANG_TIDY_IGNORED_FILES)
```
The CMake variable `3RD_PARTY_SOURCES` is not used.